platform-api/che-core-api-core/src/main/java/org/eclipse/che/everrest/DownloadFileResponseFilter.java

```diff
@@ -0,0 +1,132 @@
+/*******************************************************************************
+ * Copyright (c) 2012-2016 Codenvy, S.A.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Codenvy, S.A. - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.che.everrest;
+
+import org.eclipse.che.dto.server.JsonSerializable;
+import org.everrest.core.ApplicationContext;
+import org.everrest.core.Filter;
+import org.everrest.core.GenericContainerResponse;
+import org.everrest.core.ResponseFilter;
+import org.everrest.core.impl.ApplicationContextImpl;
+
+import javax.ws.rs.HttpMethod;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.MultivaluedMap;
+import javax.ws.rs.core.Request;
+import javax.ws.rs.core.Response;
+
+import java.util.List;
+
+import static javax.ws.rs.core.HttpHeaders.CONTENT_DISPOSITION;
+import static org.eclipse.che.everrest.DownloadFileResponseFilter.EntityType.JSON_SERIALIZABLE;
+import static org.eclipse.che.everrest.DownloadFileResponseFilter.EntityType.STRING;
+import static org.eclipse.che.everrest.DownloadFileResponseFilter.EntityType.UNKNOWN;
+/**
+ * Filter implementing {@link ResponseFilter} in order to provide the json result as a file download operation.
+ * It is applying on GET method and JSON content type only.
+ *
+ * @author Florent Benoit
+ */
+@Filter
+public class DownloadFileResponseFilter implements ResponseFilter {
+
+    /**
+     * Query parameter used to ask to specify headers that will propose JSON object to be downloaded.
+     */
+    private static final String QUERY_DOWNLOAD_PARAMETER = "downloadAsFilename";
```

Comment:
I understand your point but downloadAsFile=[name of target file] Looks OK for me.